ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1853,6 +1853,7 @@ CompletableFuture<Message> applyLogToStateMachine(ReferenceCountedObject<LogEntr
   void notifyTruncatedLogEntry(LogEntryProto logEntry) {
     if (logEntry.hasStateMachineLogEntry()) {
       getTransactionManager().remove(TermIndex.valueOf(logEntry));
+      Optional.ofNullable(getState()).ifPresent(s -> s.truncate(logEntry.getIndex()));
```

Comment:
How could `getState()` return null?  In a mock test?

Reply:
Oops, it should be outside the if-statement, i.e.
```java
+++ b/ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java
@@ -1851,6 +1851,8 @@ class RaftServerImpl implements RaftServer.Division,
    * @param logEntry the log entry being truncated
    */
   void notifyTruncatedLogEntry(LogEntryProto logEntry) {
+    getState().truncate(logEntry.getIndex());
+
     if (logEntry.hasStateMachineLogEntry()) {
       getTransactionManager().remove(TermIndex.valueOf(logEntry));
```

Reply:
@szetszwo Thanks for the quick review! Indeed the `getState()` will return null in the mock test `TestSegmentedRaftLog.testAppendEntriesWithInconsistency`, which only creates a `SegmentedRaftLog`.

Reply:
Thanks for the info!  The change looks good.